libs/route-config/src/lib/route-config.service.ts

```diff
@@ -19,40 +20,63 @@ export type RouteData<
 
 export type RouteDataParam<ConfigParamsNames extends string> = keyof RouteData<ConfigParamsNames>;
 
+const gatherRoutes = (activatedRoute: ActivatedRoute): ActivatedRoute[] => {
+  const routes: ActivatedRoute[] = activatedRoute.pathFromRoot;
+
+  let route = activatedRoute.firstChild;
+  while (route) {
+    routes.push(route);
+    route = route.firstChild;
+  }
+
+  return routes;
+};
+
 @Injectable()
 export class RouteConfigService<
   RouteTags extends string = string,
   ConfigParamsNames extends string = never
 > {
-  constructor(private activatedRoute: ActivatedRoute, private router: Router) {}
+  private get injectedDefaultValue(): Partial<RouteData<ConfigParamsNames, RouteTags>> {
+    return this._injectedDefaultValue || {};
+  }
 
-  getLeafConfig(paramName: 'routeTags', defaultValue: RouteTags[]): Observable<RouteTags[]>;
-  getLeafConfig<T>(paramName: ConfigParamsNames, defaultValue: T): Observable<T>;
-  getLeafConfig<T = unknown>(
-    paramName: RouteDataParam<ConfigParamsNames>,
-    defaultValue: T
-  ): Observable<T> {
+  constructor(
+    private activatedRoute: ActivatedRoute,
+    private router: Router,
+    @Optional()
+    @Inject(ROUTE_DATA_DEFAULT_VALUE)
+    private _injectedDefaultValue?: Partial<RouteData<ConfigParamsNames, RouteTags>>
+  ) {}
+
+  getWholeLeafConfig<
+    C extends RouteData<ConfigParamsNames, RouteTags> = RouteData<ConfigParamsNames, RouteTags>
+  >(defaultValue: Partial<C> = {}): Observable<Partial<C>> {
     return this.router.events.pipe(
       filter((event) => event instanceof ActivationEnd),
       map(() => this.activatedRoute),
       startWith(this.activatedRoute),
-      switchMap((activatedRoute: ActivatedRoute) => {
-        const routes: ActivatedRoute[] = activatedRoute.pathFromRoot;
-
-        let route = activatedRoute.firstChild;
-        while (route) {
-          routes.push(route);
-          route = route.firstChild;
-        }
-
-        return combineLatest(routes.map(({ data }) => data)).pipe(
-          map((dataArr) => {
-            const reversedArr = dataArr.reverse();
-            const index = reversedArr.findIndex((data) => (data && data[paramName]) !== undefined);
-            return (reversedArr[index] && reversedArr[index][paramName]) || defaultValue;
-          })
-        );
-      })
+      map(gatherRoutes),
+      switchMap((routes) =>
+        combineLatest(routes.map(({ data }) => data)).pipe(
```

Comment:
can `combineLatest` emit if you provide a not-observable array inside it? 🤔 

shouldn't this be `routes.map(({ data }) => of(data))` ?

Reply:
Also `combineLatest` emits every time a source observable emits, wouldn't `zip` be better here? 🤔

Reply:
Actually, the `data` property is an Observable. That's why I used `combineLatest`
I believe `combineLatest` is a proper operator in here. In general I don't think that `data` would emit more than one value. But if part of the route configuration changes for some reason I think we need to recalculate the whole config